src/proof.ml

```diff
@@ -0,0 +1,44 @@
+open Core
+open! Lplib
+open Term
+open Timed
+open ConfigBuiltin
+
+(** [is_axim s excluded] return true if [s] ends with string `_ax` and [s] is 
+    different from [excluded]. *)
+let is_axiom : string -> string -> bool = fun s excluded ->
+    if String.length s > 3 && s <> excluded then
+        String.sub s (String.length s - 3) 3 = "_ax"
+    else
+        false

```

Comment:
Can be simplified to `String.length s > 3 && s <> excluded && String.sub s (String.length s - 3) 3 = "_ax"`.